Add Azure SQL instance scoping configuration page (#36661)
DataDog/documentation: content/en/database_monitoring/architecture/azure_sql.md

---
title: Azure SQL Configurations
description: Configure Azure SQL instance scoping for Database Monitoring across SQL Database, Managed Instance, and Elastic Pool deployment types.
further_reading:
- link: "/database_monitoring/architecture/"
  tag: "Documentation"
  text: "DBM Setup Architectures"
- link: "/database_monitoring/setup_sql_server/"
  tag: "Documentation"
  text: "Setting up SQL Server"
---

## Overview

> Each check instance in the Datadog Agent configuration creates one Database Monitoring (DBM) instance. Azure SQL is a family of three deployment types, each using the SQL Server check but with different behavior.

Reviewer: ```suggestion
Each check instance in the Datadog Agent configuration creates one Database Monitoring (DBM) instance. Azure SQL is a family of three deployment types, each using the SQL Server check with different behaviors.
```

<table>
  <thead>
    <tr>
      <th style="min-width:180px">Deployment type</th>
      <th>Database Autodiscovery</th>
      <th>Notes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><code>sql_database</code></td>
      <td>Not supported</td>
      <td>One config entry per database required</td>
    </tr>
    <tr>
      <td><code>managed_instance</code></td>
      <td>Supported</td>
      <td>Behaves like on-premises SQL Server</td>
    </tr>
    <tr>
      <td>Elastic pool</td>
      <td>Not supported</td>

>       <td>Elastic pool is a billing unit only; databases connect like <code>sql_database</code></td>

Reviewer: ```suggestion
      <td>Billing unit only; one config entry per database required</td>
```

Reviewer: This is a little more direct than referencing the user back to a different entry. You mention the `sql_database` similarity later in the doc which should help make the connection

```

### Option 2: Logical server level

<div class="alert alert-info">Use logical server level only if per-database DBM data is not required. Query samples and explain plans are not collected at this scope.</div>

One entry connected to `master` covers server-level metadata only.

Based on the example scenario above, this counts as one DBM instance.

```yaml
instances:
  - host: myserver.database.windows.net
    username: datadog
    password: "<PASSWORD>"
    dbm: true
    database: master
    azure:
      deployment_type: sql_database
      fully_qualified_domain_name: myserver.database.windows.net
    database_identifier:
      template: "$azure_name"   # → "myserver"
```

## Azure SQL Managed Instance

Azure SQL Managed Instance uses engine edition 8, which supports `database_autodiscovery` and behaves like on-premises SQL Server.

**Scenario**: Managed instance `myinstance.a1b2c3d4e5f6.database.windows.net` with three databases:

| Database | Notes |
|---|---|
| `app_db` | Production application database |
| `analytics_db` | Analytics workload |
| `audit_db` | Audit records |

### Option 1: Instance level (recommended)

> <div class="alert alert-info">Use instance-level scoping (<code>$azure_name</code>). One entry with <code>database_autodiscovery: true</code> requires no manual updates as databases are added or removed.</div>

Reviewer: ```suggestion
<div class="alert alert-info">Use instance-level scoping (<code>$azure_name</code>). One entry with <code>database_autodiscovery: true</code> requires no manual updates when databases are added or removed.</div>
```
